#!/usr/bin/env python
PACKAGE = "return_local_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("odom_topic",           str_t,    0, "using for OdomHelper Class",          "/odom")

# -- PlannerLimit --
gen.add("xy_goal_tolerance",    double_t, 0, "goal tolerance as xy",                0.3, 0,  1)
gen.add("yaw_tolerance",        double_t, 0, "yaw tolerance",                       0.2, 0,  1)
gen.add("max_vel_theta",        double_t, 0, "max angular velocity",                2,   0,  10)

# -- my parameter --
gen.add("linear_vel",           double_t, 0, "linear velocity at PID",              0.2, -1, 1)
gen.add("rotate_to_goal_vel",   double_t, 0, "angular velocity for rotateToGoal",   0.3, -1, 1)

# -- PID --
gen.add("kp",      double_t, 0, "Kp value",     0.4, -10,    10)
gen.add("ki",      double_t, 0, "Ki value",     1,   -10,    10)
gen.add("kd",      double_t, 0, "Kd value",     0.7, -10,    10)
gen.add("period",  double_t, 0, "Period(hz)",   0.1, 0.0001, 10)

exit(gen.generate(PACKAGE, "return_local_planner", "ReturnLocalPlannerRos"))